src/VirtoCommerce.NativePaymentMethods.Web/Controllers/Api/NativePaymentMethodsController.cs

```diff
@@ -1,24 +1,64 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
 using VirtoCommerce.NativePaymentMethods.Core;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
+using VirtoCommerce.NativePaymentMethods.Core.Models;
+using VirtoCommerce.NativePaymentMethods.Core.Models.Search;
+using VirtoCommerce.Platform.Core.GenericCrud;
 
 
 namespace VirtoCommerce.NativePaymentMethods.Web.Controllers.Api
 {
-    [Route("api/NativePaymentMethods")]
+    [Route("api/native-payment-methods")]
     public class NativePaymentMethodsController : Controller
     {
-        // GET: api/VirtoCommerce.NativePaymentMethods
-        /// <summary>
-        /// Get message
-        /// </summary>
-        /// <remarks>Return "Hello world!" message</remarks>
+        private readonly ICrudService<NativePaymentMethod> _service;
+
+        private readonly ISearchService<PaymentMethodsSearchCriteria, PaymentMethodsSearchResult, NativePaymentMethod> _searchService;
+
+        public NativePaymentMethodsController(
+            ICrudService<NativePaymentMethod> service,
+            ISearchService<PaymentMethodsSearchCriteria, PaymentMethodsSearchResult, NativePaymentMethod> searchService)
+        {
+            _service = service;
+            _searchService = searchService;
+        }
+
         [HttpGet]
         [Route("")]
         [Authorize(ModuleConstants.Security.Permissions.Read)]
-        public ActionResult<string> Get()
+        public async Task<ActionResult<IEnumerable<NativePaymentMethod>>> GetAll()
+        {
+            return Ok(await _searchService.SearchAsync(new PaymentMethodsSearchCriteria()));
```

Comment:
What about pagination?

Reply:
We agreed that we won't have more than 10-20 payment methods.